### ISSUE TITLE
global scope javascript functions such as "remove"

### ISSUE THREAD

**Issue by photomedia** [eprints/eprints3.4#409]
I recently encountered an issue of compatibility with other plugins due to the fact that prototype is defining functions like "remove" at the global scope. 

https://github.com/eprints/eprints3.4/blob/2467b680373ac7be831e729a50c632ba7ec15e85/ingredients/prototypejs/static/javascript/auto/20_prototype.js#L2174

This causes problems when other functions also define a function like "remove".  I am referring specifically to this as an example: 

https://trustcassie.com/

which is used in my institution for cookie permissions handling.  Overall, I think that a best practice would be to contain function names like "remove" within a scope, or at least use names with a prefix or something like that.  That goes for the third-party apps too, obviously, but I can't control what they do.

**Comment by drn05r** [comment]
I will take a look at scoping.  However, ultimately PrototypeJS will be completely removed in the next major release of EPrints.

**Comment by photomedia** [comment]
Thank you for adding that to a milestone.  I realize that it might not be a simple fix, because it is a core library for EPrints now, so it could be referenced all over the place.  

"ultimately PrototypeJS will be completely removed in the next major release of EPrints"  that is great news!  

Do you mean by "next major release" EPrints 3.5?

**Comment by drn05r** [comment]
Yes, 3.5.  However, there is no release date scheduled for that yet.
